monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -0,0 +1,728 @@
+
+    For example:
+
+    .. code-block:: bash
+
+        python -m monai.apps.auto3dseg \
+            DataAnalyzer \
+            get_all_case_stats \
+            --datalist="my_datalist.json" \
+            --dataroot="my_dataroot_dir"
+
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str = "",
+        output_path: str = "./data_stats.yaml",
+        average: bool = True,
+        do_ccp: bool = True,
+        device: Union[str, torch.device] = "cuda",
+        worker: int = 2,
+        image_key: str = "image",
+        label_key: str = "label",
+    ):
+        """
+        The initializer will load the data and register the functions for data statistics gathering.
+        """
+        if path.isfile(output_path):
+            warnings.warn(f"File {output_path} already exists and will be overwritten.")
+            logger.debug(f"{output_path} will be overwritten by a new datastat.")
+        self.output_path = output_path
+        self.IMAGE_ONLY = True if label_key is None else False
+
+        if self.IMAGE_ONLY:
+            keys = [image_key]
+        else:
+            keys = [image_key, label_key]
+
+        transform_list = [
+            transforms.LoadImaged(keys=keys),
+            transforms.EnsureChannelFirstd(keys=keys),  # this creates label to be (1,H,W,D)
+            transforms.Orientationd(keys=keys, axcodes="RAS"),
+            transforms.EnsureTyped(keys=keys, data_type="tensor"),
+        ]
+
+        if not self.IMAGE_ONLY:
+            transform_list += [
+                transforms.Lambdad(
+                    keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x
+                ),
+                transforms.SqueezeDimd(keys=["label"], dim=0),
+            ]
+
+        files, _ = datafold_read(datalist=datalist, basedir=dataroot, fold=-1)
+        ds = data.Dataset(data=files, transform=transforms.Compose(transform_list))
+
+        self.dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=worker, collate_fn=no_collation)
+        # Whether to average all the modalities in the summary
+        # If SUMMARY_AVERAGE is set to false,
+        # the stats for each modality are calculated separately
+        self.SUMMARY_AVERAGE = average
+        self.DO_CONNECTED_COMP = do_ccp
+        self.device = device
+        self.data: Dict[str, Any] = {}
+        self.results: Dict[str, Any] = {}
+        # gather all summary function for combining case stats
+        self.gather_summary: Dict[str, Dict] = {}
+        self._register_functions()
+        self._register_operations()
+
+    def _register_functions(self):
+        """
+        Register all the statistic functions for calculating stats for individual cases and overall. If one installs
+        monai in the editable source code manner, then a new statistics calculation can be customized in the DataAnalyzer
```

Comment:
We should not consider the "editable source code" use case.

Thanks.

Reply:
Fixed by the refractor task